src/main/java/frc/robot/components/IMU.java

```diff
@@ -0,0 +1,130 @@
+package frc.robot.components;
+import com.kauailabs.navx.frc.AHRS;
+import edu.wpi.first.wpilibj.*;
+
+public  class IMU{
+    AHRS ahrs;
+    public double roll = 0;
+    public double pitch = 0;
+    public double yaw = 0;
+    public double x_acceleration = 0;
+    public double y_acceleration = 0;
+    public double z_acceleration = 0;
+    public double y_angularVelocity = 0;
+    public double turnRate;
+    public double maxAutoYaw = 60;// messured in degrees, may change
+
+  //the init method
+    public void init() {
+        try {
+            ahrs = new AHRS(SPI.Port.kMXP);
+            ahrs.enableLogging(true);
+        }
+
+        catch (RuntimeException ex) {
+            DriverStation.reportError("Error instantiating navX MXP:  " + ex.getMessage(), true);
+        }
+    }
+
+    //sets the yaw to zero
+    public void zeroYaw() {
+        ahrs.zeroYaw();
+        ahrs.reset();
+        yaw = 0;
+    }
+
+    //returns true if the yaw is between the valid range
+    public boolean isYawValid() {
+        if(ahrs.getYaw() < maxAutoYaw && ahrs.getYaw() > -maxAutoYaw) {
+            return true;
+        }
+
+        else {
+            return false;
+        }
+    }
+
+  //gets all the yaw values
+  public void getvalues() {
+        x_acceleration = ahrs.getWorldLinearAccelX();
+        y_acceleration = ahrs.getWorldLinearAccelY();
+        z_acceleration = ahrs.getWorldLinearAccelZ();
+        roll = ahrs.getPitch();
+        pitch = ahrs.getRoll();
+        yaw = (ahrs.getYaw()) * (Math.PI/180);// normally negated
+        turnRate = ahrs.getRate(); 
+        y_angularVelocity = ahrs.getRate();
+   }
+         
+    /*the following are all the functions avalible*/
+
+    //SmartDashboard.putNumber("IMU_Yaw", ahrs.getYaw());
+    //SmartDashboard.putNumber("IMU_Pitch", ahrs.getRoll());
+    //SmartDashboard.putNumber("IMU_Roll", ahrs.getPitch());
+    
+    //SmartDashboard.putNumber("IMU_CompassHeading", ahrs.getCompassHeading());
+
+    /* Display 9-axis Heading (requires magnetometer calibration to be useful) */
+   // SmartDashboard.putNumber("IMU_FusedHeading", ahrs.getFusedHeading());
+
+    /* These functions are compatible w/the WPI Gyro Class, providing a simple */
+    /* path for upgrading from the Kit-of-Parts gyro to the navx MXP */
+
+    // SmartDashboard.putNumber("IMU_TotalYaw", ahrs.getAngle());
+    // SmartDashboard.putNumber("IMU_YawRateDPS", ahrs.getRate());
```

Comment:
It might be good to put TotalYaw and YawRateDPS on the dashboard while we are debugging